### ISSUE TITLE
Installation woes

### ISSUE THREAD

**Issue by Newres** [dvcrn/proton#234]
One issue that I ran into a few times when I aimed to try out Proton is how heavy and sluggish the install feels. 

It takes a really long time before Proton as a whole is installed,, and in the meanwhile Atom is completely unresponsive.  Even the list of what is being installed goes halfway outside the window midway during install..

I tried the install multiple times on two different machines, both from the published version and from the source, but I still had this issue..

I was wondering if this was the normal install experience or if I ran into some issue that I do not know about.

**Comment by dvcrn** [comment]
That is indeed a problem but also not easy to solve. 

proton can only fully work when all packages have been installed and applied. Before that if we allow using the editor normal, you might experience some very weird behavior. 

One idea would be to just deny all SPC commands until initialization is done and move the window into one of the corners without the modal block. Would that help with this issue for you? 

Pinging @geksilla for thoughts

**Comment by geksilla** [comment]
I'm agree that installation process takes a lot of time. Probably we can make installation in background, do not block user on editing and prioritize package/layer installation.
A lot of layers defines keybindings under appropriate package so they will appear in *proton-modal* only after package become installed.
I can take a look how we can optimize this case, but I don't promise to resolve this ASAP. Unfortunately I'm limited on time.

**Comment by Newres** [comment]
I think perhaps a possible solution would be if instead of current checklist  of installed packages I could see a terminal, with a more verbose output. I had a couple of installs when atom hanged, or felt like it hanged, but could not tell for sure either way. And if something goes wrong during install it would help to pinpoint the issue.

**Comment by virginc** [comment]
I would be happy if I could install it from a terminal window.

**Comment by dvcrn** [comment]
@virginc you can run `apm install proton-mode` to install proton, but this will not take care of the dependency installation triggered through proton

**Comment by armed** [comment]
+1 for shell script with all apm installs in it.

**Comment by geksilla** [comment]
We may use status bar instead of modal panel to display installation progress. I think that good approach will be using [busy-signal](https://atom.io/packages/busy-signal) package.
What do you think?

**Comment by agustif** [comment]
Yup, I think busy-signal and a discrete load is the best approach, after all you don't need to block the ui until all the dependencies are installed and ready to load. The -normal- or default editor is enough
